examples/smartcore-ml/README.md

```diff
@@ -0,0 +1,119 @@
+```
+
+```
+let model: GaussianNB<{float}, _, DenseMatrix<{float}>, _> = trained_model;
+```
+
+```
+let model: LogisticRegression<{flaot}, u32, DenseMatrix<{float}>, Vec<u32>> = trained_model;
+```
+
+```
+let model: LinearRegression< {float}, {float}, DenseMatrix<{float}>, Vec<{float}>> = trained_model;
+```
+
+```
+let model: KNNClassifier< {float}, u32, DenseMatrix<{float}>, Vec<u32>, Euclidian<{float}>> = trained_model;
+```
+
+```
+let model: KMeans<{float}, u8, DenseMatrix<{float}>, Vec<u8>> = trained_model;
+```
+
+```
+let model: PCA<{float}, DenseMatrix<{float}>> = trained_model;
+```
+
+NOTE:  For Support Vector Machine classifiers and regression models, you MUST use the following smartcore fork and import into you cargo.toml file:
+
+smartcore = { git = "https://github.com/Roee-87/smartcore.git", features = ["serde"]}
+
+Deserialization of SVC and SVR does not includes that SVCParametersmake field.  It must be added back into the model struct manually after deserialization.  The SVC and SVR fork have ammended the visbility of the model struct, making the parameters field public and thereby allowing the parameters field to be directly inserted into the model struct after dserialization.  You must insert the same parameters that were used when training the model.
+```
+let mut model: SVC<{float}, i32, DenseMatrix<{float}>, Vec<i32>> = deserialized_svc_trained;
+let params = &SVCParameters::default().with_c(200.0).with_kernel(Kernels::linear());
+model.parameters = Some(params);
+```
+
+```
+let mut model: SVR< '_, {float}, DenseMatrix<{float}>, Vec<{float}>> = deserialized_model;
+//make sure to use the same SVRParameters used to train the model
+let params = &SVRParameters::default().with_eps(2.0).with_c(10.0).with_kernel(Kernels::linear());
+model.parameters = params;
+```
+
+SVM models support four kernels:  Gaussian (rbf), Linear, Polynomial, and Sigmoid.  Kernels are selected within the .with_kernel() function in the SVCParameters or SVRParameters.
+Kernel::linear()
+Kernel::rbf()
+Kernel::polynomial()
+Kernel::sigmoid()
+
+## Quick Start
+
+First, make sure [rustup](https://rustup.rs) is installed. This project uses a [nightly](https://doc.rust-lang.org/book/appendix-07-nightly-rust.html) version of [Rust](https://doc.rust-lang.org/book/ch01-01-installation.html). The [`rust-toolchain`](rust-toolchain) file will be used by `cargo` to automatically install the correct version.
+
+To build all methods and execute the method within the zkVM, run the following command:
+
+```
+cargo run
+```
+### Running proofs remotely on Bonsai
+
+*Note: The Bonsai proving service is still in early Alpha; an API key is required for access. [Click here to request access].*
+
+If you have access to the URL and API key to Bonsai you can run your proofs
+remotely. To prove in Bonsai mode, invoke `cargo run` with two additional
+environment variables:
+
+```
+BONSAI_API_KEY="YOUR_API_KEY" BONSAI_API_URL="BONSAI_URL" cargo run
```

Comment:
If you use `< >` then people who forget to substitute get shell errors rather than something potentially more mysterious.
```suggestion
BONSAI_API_KEY=<YOUR_API_KEY> BONSAI_API_URL=<BONSAI_URL> cargo run
```

Reply:
Removing this section.